src/handle_trailer.sh

```diff
@@ -0,0 +1,169 @@
+    complain 'An option is required to use this command.'
+    handle_trailer_help
+    return 22 # EINVAL
+  fi
+
+  for arg in "${PATCH_OR_SHA[@]}"; do
+    # Check if given argument is either a patch or valid commit reference,
+    # then build the correct command.
+    if [[ $(git cat-file -t "$arg" 2> /dev/null) == 'commit' ]]; then
+      cmd="git commit --quiet --amend --no-edit --trailer \"${TRAILER_TAG} ${TRAILER_VALUE}\""
+      # Only call 'git rebase' if user is trying to handle multiple commits
+      if [[ "$(git rev-parse "$arg")" != "$(git rev-parse HEAD)" ]]; then
+        cmd="git rebase ${arg} --exec '${cmd}' 2> /dev/null"
+      fi
+    elif is_a_patch "$arg"; then
+      cmd="git interpret-trailers ${arg} --in-place --trailer"
+      cmd+=" \"${TRAILER_TAG} ${TRAILER_VALUE}\""
+    else
+      warning "Neither a patch nor a valid commit. Ignoring ${arg}"
+      continue
+    fi
+    cmd_manager "$flag" "$cmd"
+  done
+}
+
+# This function gets raw data and based on that fill out the options values to
+# be used in another function.
+#
+# Return:
+# In case of successful return 0, otherwise, return 22.
+function parse_handle_trailer_options()
+{
+  local long_options='add-reviewed-by:,add-acked-by:,add-fixes:,verbose'
+  local short_options='r:,a:,f:'
+
+  options="$(kw_parse "$short_options" "$long_options" "$@")"
+
+  if [[ "$?" != 0 ]]; then
+    options_values['ERROR']="$(kw_parse_get_errors 'kw commit-trailer' \
+      "$short_options" "$long_options" "$@")"
+    return 22 # EINVAL
+  fi
+
+  options_values['TRAILER_TAG']=''
+  options_values['TRAILER_VALUE']=''
+  options_values['PATCH_OR_SHA']='HEAD'
+  options_values['VERBOSE']=''
+
+  eval "set -- ${options}"
+
+  while [[ "$#" -gt 0 ]]; do
+    case "$1" in
+      --add-reviewed-by | -r)
+        options_values['TRAILER_TAG']='Reviewed-by:'
+        options_values['TRAILER_VALUE']="$(str_strip "$2")"
+        shift 2
+        ;;
+      --add-acked-by | -a)
+        options_values['TRAILER_TAG']='Acked-by:'
+        options_values['TRAILER_VALUE']="$(str_strip "$2")"
+        shift 2
+        ;;
+      --add-fixes | -f)
+        options_values['TRAILER_TAG']='Fixes:'
+        options_values['TRAILER_VALUE']="$(str_strip "$2")"
+
+        # Check if given value is a valid commit reference
+        if [[ $(git cat-file -t "${options_values['TRAILER_VALUE']}" 2> /dev/null) != 'commit' ]]; then
+          options_values['ERROR']='Invalid commit reference with --add-fixes option: '
+          options_values['ERROR']+="${options_values['TRAILER_VALUE']}"
+          return 22 # EINVAL
+        fi
+        # The 'Fixes:' trailer line must follow a format defined by
+        # Linux Kernel developers. Example:
+        # Fixes: e21d2170f366 ("video: remove unnecessary platform_set_drvdata()")
+        options_values['TRAILER_VALUE']=$(git log -1 "${options_values['TRAILER_VALUE']}" \
+          --oneline --format="%h (\\\"%s\\\")" \
+          --abbrev-commit --abbrev=12)
```

Comment:
It would be better if this manipulation were moved to a single function and called from the main functions instead of here.